### PR TITLE
Update software release policy for osg 3.6

### PR DESCRIPTION
**Pull request by timtheisen**

### CODE REVIEW

**Comment by brianhlin** [comment]
Marking `do-not-merge` since we want to hold off until we're closer to the 3.6 release

File: docs/policy/software-release.md [hunk 1]
```diff
@@ -14,20 +14,28 @@ The Software Team maintains the following Yum repositories:
...
+
+We also create a repository per release, called **osg-release-VERSION** (such as osg-release-3.5.4).
+This is intended mostly for testing purposes, though users may occasionally find it useful.
+
 Version Numbers
 ---------------
 
+!!! note "OSG 3.5 Only"
+    This version number scheme is only applicable to the OSG 3.5 series.
+
```
Comment (brianhlin) [inline docs/policy/software-release.md:38]: Looking at this section, maybe we should make this a sub-section and add a separate `3.6+` section that just says that the version numbers match the release series?

File: docs/policy/software-release.md [hunk 2]
```diff
@@ -42,24 +50,20 @@ However, there are important caveats about the version number:
...
+The version number will be of the form X.Y.Z. As of this writing, version numbers are 3.5.Z, where Z indicates a minor
 revision.
 
 Progression of Repositories
 ---------------------------
 
 This figure shows the progression of repositories that packages will go through:
 
-     osg-development -> osg-testing -> osg-prerelease / osg-rolling -> osg-release
+     osg-development -> osg-testing -> osg-prerelease -> osg-release
```
Comment (brianhlin) [inline docs/policy/software-release.md:66]: Since we communicate above that 3.6 doesn't have a rolling repo, I think we can keep this

**Review by brianhlin — APPROVED** [review]
Minor suggestions but otherwise LGTM

**Comment by matyasselmeci** [comment]
This deletes a lot of information about osg-rolling; perhaps that could be moved into a '3.5 only' section instead? After all, we're still going to follow the usual policy about rolling and prerelease until 3.5 is completely dropped...

File: docs/policy/software-release.md [hunk 3]
```diff
@@ -14,20 +14,28 @@ The Software Team maintains the following Yum repositories:
...
 
+!!! note
+    **osg-rolling** and **osg-release-VERSION** are only present in the OSG 3.5 series.
+
+1.  **osg-rolling**: This is where software goes before being included in a point release. Intended for end-users.
+    In OSG 3.5, software goes into **osg-rolling** when it is put into **osg-prerelease**.
+
+We also create a repository per release, called **osg-release-VERSION** (such as osg-release-3.5.4).
+This is intended mostly for testing purposes, though users may occasionally find it useful.
+
```
[the remark this answers is not in the document]
Comment (timtheisen) [reply docs/policy/software-release.md]: Updated as requested.

File: docs/policy/software-release.md (hunk 2 above)
Comment (timtheisen) [reply docs/policy/software-release.md]: Updated as requested

File: docs/policy/software-release.md (hunk 1 above)
Comment (timtheisen) [reply docs/policy/software-release.md]: Updated as requested

**Comment by timtheisen** [comment]
CI failure was transient external web site failure.

**Pull request merged**